code/datums/gamemodes/campaign/missions/base_rescue.dm

```diff
@@ -145,6 +145,11 @@
 	else
 		icon_state = "[initial(icon_state)]_open"
 
+/obj/structure/weapon_x_pod/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.loc != src)
```

Comment:
```suggestion
	if(X != occupant)
	    return
```

This feels like it describes the check better, making things simpler to understand